core/spec/models/spree/promotion_spec.rb

```diff
@@ -64,6 +65,10 @@
     it "should delete rules" do
       expect(Spree::PromotionRule.count).to eq(0)
     end
+
+    it "should delete codes" do
+      expect(Spree::PromotionCode.count).to eq(0)
```

Comment:
A nice-to-have: I think this might be slightly better as `expect(Spree::PromotionCode.where(promotion_id: promotion.id).count).to eq(0)`.  (And the same thing in the other specs above this one.)  Though I also am not sure that we even need to test that `dependent: destroy` does what it says it'll do.